.pipelines/.vsts-vhd-builder-release-windows.yaml

```diff
@@ -162,12 +175,53 @@ stages:
             parameters:
               artifactName: 23H2-gen2
 
-  - stage: e2e
-    condition: ne(variables.SKIP_E2E_TESTS, 'true')
+  - stage: e2e_win_2019_gen1
+    dependsOn: [ build_win_2019_gen1 ]
+    condition: and(succeeded(), ne(variables.SKIP_E2E_TESTS, 'true'))
+    variables:
+      VHD_BUILD_ID: $(Build.BuildId)
+      LOCATION: $(PACKER_BUILD_LOCATION)
+      TAGS_TO_RUN: "os=windows,imageName=windows-2019-containerd"
+    jobs:
+      - template: ./templates/e2e-template.yaml
+
+  - stage: e2e_win_2022_gen1
+    dependsOn: [ build_win_2022_gen1 ]
+    condition: and(succeeded(), ne(variables.SKIP_E2E_TESTS, 'true'))
+    variables:
+      VHD_BUILD_ID: $(Build.BuildId)
+      LOCATION: $(PACKER_BUILD_LOCATION)
+      TAGS_TO_RUN: "os=windows,imageName=windows-2022-containerd"
+    jobs:
+      - template: ./templates/e2e-template.yaml
+
+  - stage: e2e_win_2022_gen2
+    dependsOn: [ build_win_2022_gen2 ]
+    condition: and(succeeded(), ne(variables.SKIP_E2E_TESTS, 'true'))
+    variables:
+      VHD_BUILD_ID: $(Build.BuildId)
+      LOCATION: $(PACKER_BUILD_LOCATION)
+      TAGS_TO_RUN: "os=windows,imageName=windows-2022-containerd-gen2"
+    jobs:
+      - template: ./templates/e2e-template.yaml
+
+  - stage: e2e_win_23H2_gen1
+    dependsOn: [ build_win_23H2_gen1 ]
```

Comment:
Maybe it will look cleaner if it will be part of the build stage?
This way you won't need to specify dependsOn

Reply:
I tried that and it didn't work - probably due to the way the template is written.